src/git.js

```diff
@@ -242,19 +242,49 @@ class Git {
 		const tree = []
 		for (const treeObject of output.split('\n')) {
 			const [ mode, type, sha ] = treeObject.split(/\s/)
-			const treeObjectPath = treeObject.split('\t')[1]
+			const file = treeObject.split('\t')[1]
 
 			const treeEntry = {
 				mode,
 				type,
-				content: await this.getBlobContent(sha),
-				path: treeObjectPath
+				sha,
+				path: file
 			}
+
 			tree.push(treeEntry)
 		}
+
 		return tree
 	}
 
+	// Creates the blob objects in GitHub for the files that are not in the previous commit only
+	async createGithubBlobs(commitSha) {
+		core.debug('Creating missing blobs on GitHub')
+		const [ previousTree, tree ] = await Promise.all([ this.getTree(`${ commitSha }~1`), this.getTree(commitSha) ])
+		const promisesGithubCreateBlobs = []
+
+		for (const treeEntry of tree) {
+			// If the current treeEntry are in the previous tree, that means that the blob is uploaded and it doesn't need to be uploaded to GitHub again.
+			if (previousTree.findIndex((entry) => entry.sha === treeEntry.sha) !== -1) {
+				continue
+			}
+
+			const base64Content = await this.getBlobBase64Content(treeEntry.path)
+
+			// Creates the blob. We don't need to store the response because the local sha is the same and we can use it to reference the blob
+			const githubCreateBlobRequest = this.github.git.createBlob({
+				owner: this.repo.user,
+				repo: this.repo.name,
+				content: base64Content,
+				encoding: 'base64'
+			})
+			promisesGithubCreateBlobs.push(githubCreateBlobRequest)
+		}
+
+		// Wait for all the file uploads to be completed
+		await Promise.all(promisesGithubCreateBlobs)
+	}
```

Comment:
This function compares the content of the tree to commit with the previous tree.

Iterates over each blob entry for the current commit checking if it was present on the previous tree. If it's not present, it uploads it to GitHub, if not just skip the upload.